implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java

```diff
@@ -0,0 +1,362 @@
+
+        if (config.uri().matches("/metrics/ingest/?$")) {
+            endpoint = config.uri();
+        } else {
+            try {
+                endpoint = new URL(new URL(config.uri()), "/api/v2/metrics/ingest").toString();
+            } catch (MalformedURLException e) {
+                logger.warn("could not parse endpoint url. Falling back to local OneAgent endpoint.");
+                endpoint = "http://127.0.0.1:14499/metrics/ingest";
+            }
+        }
+        return endpoint;
+    }
+
+    private DimensionList parseDefaultDimensions(Map<String, String> defaultDimensions) {
+        if (defaultDimensions == null || defaultDimensions.isEmpty()) {
+            return DimensionList.create(Dimension.create("dt.metrics.source", "micrometer"));
+        }
+
+        // add the dynatrace default dimension
+        Stream<Map.Entry<String, String>> concatenated = Stream.concat(
+                defaultDimensions.entrySet().stream(),
+                staticDimensions.entrySet().stream()
+        );
+
+        return DimensionList.fromCollection(concatenated.map(
+                (kv) -> Dimension.create(kv.getKey(), kv.getValue())
+        ).collect(Collectors.toList()));
+    }
+
+    @Override
+    public void export(MeterRegistry registry) {
+        // split the lines by whether or not they exceed the line length limit.
+        Map<Boolean, List<String>> metricLines =
+                registry.getMeters().stream()
+                        .flatMap(this::toMetricLines)
+                        .collect(Collectors.partitioningBy(DynatraceExporterV2::lineLengthBelowLimit));
+
+        // both keys will be present, even if empty.
+        if (!metricLines.get(false).isEmpty()) {
+            logger.info("dropping {} lines that exceed the line length limit of {}", metricLines.get(false).size(), METRIC_LINE_MAX_LENGTH);
+        }
+
+        sendInBatches(metricLines.get(true));
+    }
+
+    private Stream<String> toMetricLines(Meter meter) {
+        return meter.match(
+                this::toGaugeLine,
+                this::toCounterLine,
+                this::toTimerLine,
+                this::toDistributionSummaryLine,
+                this::toLongTaskTimerLine,
+                this::toTimeGaugeLine,
+                this::toFunctionCounterLine,
+                this::toFunctionTimerLine,
+                this::toMeterLine
+        );
+    }
+
+    Stream<String> toGaugeLine(Gauge meter) {
+        return toGauge(meter);
+    }
+
+    Stream<String> toCounterLine(Counter meter) {
+        return toCounter(meter);
+    }
+
+    double minFromHistogramSnapshot(HistogramSnapshot histogramSnapshot) {
+        ValueAtPercentile[] valueAtPercentiles = histogramSnapshot.percentileValues();
+        double min = Double.NaN;
+
+        for (ValueAtPercentile valueAtPercentile : valueAtPercentiles) {
+            if (valueAtPercentile.percentile() == 0.0) {
+                min = valueAtPercentile.value();
```

Comment:
Would it be possible to directly access the 0.0 percentile? Otherwise you could break from the for loop early since it will likely be the very first match.

Reply:
There seems to be no function that directly returns the value at a given percentile. Chenged it to break out of the loop.